### PR TITLE
EntityManager: Remove DBAL Connection factory

### PR DESCRIPTION
**Pull request by Majkl578**
Removes the hardcoded magic behind creating Connection instance from an array.

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Upgrade to 3.0
 
+## BC Break: Removed DBAL Connection factory from EntityManager
+
+EntityManager no longer provides a factory for DBAL Connection from an array.
+You must now pass an existing Connection instance directly.
+You can use `Doctrine\DBAL\DriverManager::getConnection()` factory to create new Connection.
```
Comment (Ocramius) [inline UPGRADE.md]: A better migration path is needed here. An example of before/after is needed.

Comment (Majkl578) [reply UPGRADE.md]: Updated.

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

**Comment by Ocramius** [comment]
Manually merged via doctrine/doctrine2@50da42f2f472a705586a2803f9acbb991b8db0a2, thanks!

**Pull request closed without merging**